tests/pytests/test_acl.py

```diff
@@ -8,88 +8,114 @@ def test_acl_category(env):
-    env.expect('AUTH', 'test', '123').true()
-
-    # `test` should now be able to run `write` commands
-    for cmd in WRITE_SEARCH_COMMANDS:
-        env.expect(cmd).error().notContains(
-            "User test has no permissions")
-
-    # Add `test` `search` permissions
-    env.expect('AUTH', 'default', '').true()
-    env.expect('ACL', 'SETUSER', 'test', '+@search').ok()
-    env.expect('AUTH', 'test', '123').true()
-
-    # `test` should now be able to run `search` commands like `FT.CREATE`
-    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'txt', 'TEXT').ok()
-    env.expect('FT.SEARCH', 'idx', '*').equal([0])
+
+    with env.getClusterConnectionIfNeeded() as conn:
+        # Create a user with no command permissions (full keyspace and pubsub access)
+        conn.execute_command('ACL', 'SETUSER', 'test', 'on', '>123', '~*', '&*')
+        res = conn.execute_command('AUTH', 'test', '123')
+        env.assertEqual(res, True)
+
+        # Such a user shouldn't be able to run any RediSearch commands (or any other commands)
+        try:
+            conn.execute_command('FT.SEARCH', 'idx', '*')
+            env.assertTrue(False)
+        except Exception as e:
+            env.assertEqual("User test has no permissions to run the 'FT.SEARCH' command", str(e))
+
+        # Add `test` read permissions
+        conn.execute_command('AUTH', 'default', '')
+        conn.execute_command('ACL', 'SETUSER', 'test', '+@read')
+        conn.execute_command('AUTH', 'test', '123')
+
+        # `test` should now be able to run `read` commands like `FT.SEARCH', but not
+        # `search` commands like `FT.CREATE`
+        for cmd in READ_SEARCH_COMMANDS:
+            try:
+                conn.execute_command(cmd)
+            except Exception as e:
+                env.assertNotContains("User test has no permissions to run", str(e))
+
+        # `test` should not be able to run `search` commands that are not `read`,
+        # like `FT.CREATE`
+        try:
+            conn.execute_command('FT.CREATE', 'idx', '*')
+            env.assertTrue(False)
+        except Exception as e:
+            env.assertContains("User test has no permissions to run the 'FT.CREATE' command", str(e))
+
+        # Let's add `write` permissions to `test`
+        conn.execute_command('AUTH', 'default', '')
+        conn.execute_command('ACL', 'SETUSER', 'test', '+@write')
+        conn.execute_command('AUTH', 'test', '123')
+
+        # `test` should now be able to run `write` commands
+        for cmd in WRITE_SEARCH_COMMANDS:
+            try:
+                res = conn.execute_command(cmd)
+                env.assertTrue(False)
+            except Exception as e:
+                env.assertNotContains("User test has no permissions", str(e))
+
+        # Add `test` search permissions
+        conn.execute_command('AUTH', 'default', '')
+        conn.execute_command('ACL', 'SETUSER', 'test', '+@search', '+@_search_internal')
```

Comment:
What about failing to run internal commands without `_search_internal` category?